src/main/java/uk/gov/hmcts/cmc/claimstore/events/claim/ClaimCreatedOperationHandler.java

```diff
@@ -0,0 +1,170 @@
+                : claim;
+
+        notifyRepresentativeOperation = (claim, event, docs) ->
+            isNotifyCitizenSuccess.test(claim.getClaimSubmissionOperationIndicators())
+                ? claimantOperationService.confirmRepresentative(claim, event.getSubmitterName(),
+                ((RepresentedClaimCreatedEvent) event).getRepresentativeEmail(), event.getAuthorisation())
+                : claim;
+    }
+
+    @EventListener
+    public void citizenIssueHandler(CitizenClaimCreatedEvent event) {
+        try {
+            Claim claim = event.getClaim();
+            String authorisation = event.getAuthorisation();
+            Claim updatedClaim;
+
+            GeneratedDocuments generatedDocuments = documentGenerationService.generateForCitizen(claim, authorisation);
+
+            updatedClaim = CompletableFuture
+                .supplyAsync(() -> performPinOperations.perform(claim, event, generatedDocuments))
+                .thenApplyAsync(pinCompletedClaim
+                    -> uploadSealedClaimOperation.perform(pinCompletedClaim, event, generatedDocuments))
+                .thenApplyAsync(uploadedSealedClaim
+                    -> uploadClaimIssueReceiptOperation.perform(uploadedSealedClaim, event, generatedDocuments))
+                .thenApplyAsync(claimAfterReceiptUpload
+                    -> rpaOperation.perform(claimAfterReceiptUpload, event, generatedDocuments))
+                .thenApplyAsync(claimAfterRpa
+                    -> notifyCitizenOperation.perform(claimAfterRpa, event, generatedDocuments))
+                .get();
+
+            claimService.updateClaimState(authorisation, updatedClaim, ClaimState.ISSUED);
+
+        } catch (Exception e) {
+            logger.error("failed operation processing for event ()", event, e);
+        }
+    }
+
+    @EventListener
+    public void representativeIssueHandler(RepresentedClaimCreatedEvent event) {
+        try {
+            Claim claim = event.getClaim();
+            String authorisation = event.getAuthorisation();
+
+            GeneratedDocuments generatedDocuments = documentGenerationService.generateForRepresentative(claim);
+
+            Claim updatedClaim = CompletableFuture
+                .supplyAsync(() -> uploadSealedClaimOperation.perform(claim, event, generatedDocuments))
+                .thenApplyAsync(claimAfterReceiptUpload
+                    -> rpaOperation.perform(claimAfterReceiptUpload, event, generatedDocuments))
+                .thenApplyAsync(claimAfterRpa
+                    -> notifyStaffOperation.perform(claimAfterRpa, event, generatedDocuments))
+                .thenApplyAsync(claimAfterStaffNotify
+                    -> notifyRepresentativeOperation.perform(claimAfterStaffNotify, event, generatedDocuments)
+                ).get();
+
+            claimService.updateClaimState(authorisation, updatedClaim, ClaimState.ISSUED);
```

Comment:
Should we use same state names as we define in our CCD definition? We have `OPEN` state which means claim is issued. Unless we are renaming `OPEN` to `ISSUED` in CCD definition as well.

Reply:
This will be taken care on 5374 which is CCD implementation.